util/Pos/Util/Log/LogSafe.hs

```diff
@@ -0,0 +1,397 @@
+import qualified Language.Haskell.TH as TH
+
+import           Pos.Util.Log (CanLog (..), LogContext, LoggingHandler,
+                     Severity (..), WithLogger)
+import           Pos.Util.Log.Internal (getConfig, getLogEnv, sev2klog)
+import           Pos.Util.LoggerConfig (LogHandler (..), LogSecurityLevel (..),
+                     lcLoggerTree, lhName, ltHandlers)
+
+import qualified Katip as K
+import qualified Katip.Core as KC
+
+----------------------------------------------------------------------------
+-- Logging
+----------------------------------------------------------------------------
+
+-- | Modifies logging so that it writes only to files specified by @s@ type.
+newtype SelectiveLogWrapped s m a = SelectiveLogWrapped
+    { getSecureLogWrapped :: m a
+    } deriving (Functor, Applicative, Monad, MonadIO)
+
+instance MonadTrans (SelectiveLogWrapped s) where
+    lift = SelectiveLogWrapped
+
+-- | Whether to log to given log handler.
+type SelectionMode = LogSecurityLevel -> Bool
+
+selectPublicLogs :: SelectionMode
+selectPublicLogs = \case
+    PublicLogLevel -> True
+    _              -> False
+
+selectSecretLogs :: SelectionMode
+selectSecretLogs = not . selectPublicLogs
+
+-- | conditionally log a message to all scribes if they match 'SelectionMode'
+logMCond :: (LogContext m) => LoggingHandler -> Severity -> Text -> SelectionMode -> m ()
+logMCond lh sev msg cond = do
+    ctx <- K.getKatipContext
+    ns  <- K.getKatipNamespace
+    logItemS lh ctx ns Nothing (sev2klog sev) cond $ K.logStr msg
+
+-- | this emulates katip's 'logItem' function, but only outputs the message
+--   to scribes which match the 'SelectionMode'
+logItemS
+    :: (K.LogItem a, K.Katip m)
+    => LoggingHandler
+    -> a
+    -> K.Namespace
+    -> Maybe TH.Loc
+    -> K.Severity
+    -> SelectionMode
+    -> K.LogStr
+    -> m ()
+logItemS lhandler a ns loc sev cond msg = do
+    mayle <- liftIO $ getLogEnv lhandler
+    case mayle of
+        Nothing              -> error "logging not yet initialized. Abort."
+        Just le@K.LogEnv{..} -> do
+            maycfg <- liftIO $ getConfig lhandler
+            let cfg = case maycfg of
+                    Nothing -> error "No Configuration for logging found. Abort."
+                    Just c  -> c
+            liftIO $ do
+                item <- K.Item
+                    <$> pure (K._logEnvApp le)
+                    <*> pure (K._logEnvEnv le)
+                    <*> pure sev
+                    <*> (KC.mkThreadIdText <$> myThreadId)
+                    <*> pure (K._logEnvHost le)
+                    <*> pure (K._logEnvPid le)
+                    <*> pure a
+                    <*> pure msg
+                    <*> (K._logEnvTimer le)
+                    <*> pure ((K._logEnvApp le) <> ns)
+                    <*> pure loc
+                -- forM_ (filter cond (elems (K._logEnvScribes le))) $ \ KC.ScribeHandle {..} -> atomically (KC.tryWriteTBQueue KC.shChan (NewItem item))
```

Comment:
Commented out code that I cannot even view because the line length exceeds the Github display?